tools/dds/dds-adapter/rs-dds-adapter.cpp

```diff
@@ -58,10 +60,67 @@ topics::device_info rs2_device_to_info( rs2::device const & dev )
 }
 
 
+static nlohmann::json load_config()
```

Comment:
This makes my point about a utility stronger :-)

The 3 static functions should probably all be in `rsutils`

Reply:
Two. I thought the same but I don't think it belongs in rsutils.
Load config shouldn't know about RS2_CONFIG_FILENAME. It can get the filename. But then what're we adding? The exception handling? The error would be different, I could make it "failed to load json file: ..." but then I just don't feel the function adds much.
Same with merge_settings: it's just a wrapper around `merge_patch` with error handling.
In the end I decided to just keep as is. You think I should make `rsutils::json::load_file()` and `rsutils::json::merge`?

Reply:
`load_file` in rsutils makes less and less sense to me (no filename generation; if would have to return bool to denote if the file was there, so you'd have to handle it in the caller). I prefer to just embed directly in load_settings.

`merge_patch` is a weird name and I can see introducing `rsutils::json::merge()` or `json::patch` with maybe an optional name for error handling.

Reply:
Anyway:
* added `rsutils::json::patch()`
* added `rsutils::json::load_settings` that does the basic global+app merge